ui/src/app/utils/runtime-utils.tsx

```diff
@@ -1,29 +1,219 @@
 import {runtimeApi} from 'app/services/swagger-fetch-clients';
-import {switchCase} from 'app/utils';
-import { withAsyncErrorHandling } from 'app/utils';
-import {
-  ExceededActionCountError,
-  LeoRuntimeInitializationAbortedError,
-  LeoRuntimeInitializer,
-} from 'app/utils/leo-runtime-initializer';
-import {
-  compoundRuntimeOpStore,
-  markCompoundRuntimeOperationCompleted,
-  registerCompoundRuntimeOperation,
-  runtimeStore,
-  useStore
-} from 'app/utils/stores';
-import {Runtime, RuntimeStatus} from 'generated/fetch';
+import {switchCase, withAsyncErrorHandling} from 'app/utils';
+import {ExceededActionCountError, LeoRuntimeInitializationAbortedError, LeoRuntimeInitializer, } from 'app/utils/leo-runtime-initializer';
+import {compoundRuntimeOpStore, markCompoundRuntimeOperationCompleted, registerCompoundRuntimeOperation, runtimeStore, useStore} from 'app/utils/stores';
+import {DataprocConfig, Runtime, RuntimeStatus} from 'generated/fetch';
 import * as fp from 'lodash/fp';
 
 import * as React from 'react';
+import {ComputeType, findMachineByName, Machine} from './machines';
 
 const {useState, useEffect} = React;
 
 export enum RuntimeStatusRequest {
   Delete = 'Delete'
 }
 
+export interface RuntimeDiff {
+  desc: string;
+  previous: string;
+  new: string;
+  differenceType: RuntimeDiffState;
+}
+
+export enum RuntimeDiffState {
+  NO_CHANGE,
+  CAN_UPDATE,
+  NEEDS_DELETE
+}
+
+export interface RuntimeConfig {
+  computeType: ComputeType;
+  machine: Machine;
+  diskSize: number;
+  dataprocConfig: DataprocConfig;
+}
+
+const compareComputeTypes = (oldRuntime: RuntimeConfig, newRuntime: RuntimeConfig): RuntimeDiff => {
+  return {
+    desc: 'Change compute type',
+    previous: oldRuntime.computeType,
+    new: newRuntime.computeType,
+    differenceType: oldRuntime.computeType === newRuntime.computeType ?
+      RuntimeDiffState.NO_CHANGE : RuntimeDiffState.NEEDS_DELETE
+  };
+};
+
+const compareMachineCpu = (oldRuntime: RuntimeConfig, newRuntime: RuntimeConfig): RuntimeDiff => {
+  const oldCpu = oldRuntime.machine.cpu;
+  const newCpu = newRuntime.machine.cpu;
+
+  return {
+    desc: (newCpu < oldCpu ?  'Decrease' : 'Increase') + ' number of CPUs',
+    previous: oldCpu.toString(),
+    new: newCpu.toString(),
+    differenceType: oldCpu === newCpu ? RuntimeDiffState.NO_CHANGE : RuntimeDiffState.NEEDS_DELETE
```

Comment:
The update API supports changing the machine type, so why does this require a delete?